api/handlers.go

```diff
@@ -330,6 +330,39 @@ func getVolume(c *context, w http.ResponseWriter, r *http.Request) {
 	httpError(w, fmt.Sprintf("No such volume: %s", name), http.StatusNotFound)
 }
 
+func GetUsedVolumes(containers []*cluster.Container) map[string]bool {
+	usedVolumes := make(map[string]bool)
+	for _, container := range containers {
+		for _, mount := range container.Mounts {
+			if mount.Type == "volume" {
+				usedVolumes[container.Engine.Name+"/"+mount.Name] = true
+			}
+		}
+	}
+	return usedVolumes
+}
+
+func GetFilteredVolumes(containers []*cluster.Container, volumes []*apitypes.Volume, dangling bool) []*apitypes.Volume {
+	usedVolumes := GetUsedVolumes(containers)
+	var filteredVolumes []*apitypes.Volume
+	// volume name allowed char [a-zA-Z0-9][a-zA-Z0-9_.-]
+	// splitting on / is safe to keep the original output intact
+	stripEngine := func(vol *apitypes.Volume) *apitypes.Volume {
+		if vol.Driver != "local" {
```

Comment:
Instead of special-casing local volumes, can you just do `parts = strings.Split(vol.Name)` and use `parts[1]` if `len(parts) > 1` and `vol.Name` otherwise?

Reply:
I have to special case `local` volumes so I don't change its `engine_name/volume_name` format.

Reply:
In that case, add a comment.

Reply:
done